### ISSUE TITLE
[Bug]: Generic SSO user attributes

### ISSUE THREAD

**Issue by Lunik** [BerriAI/litellm#2130]
### What happened?

Using the Generic SSO configuration provided in [the UI documentation](https://docs.litellm.ai/docs/proxy/ui#set-admin-view-w-sso)

I have set the 5 basics parameters :
```bash
export GENERIC_CLIENT_ID=XXXXXX
export GENERIC_CLIENT_SECRET=XXXXXX
export GENERIC_TOKEN_ENDPOINT=XXXXXX
export GENERIC_USERINFO_ENDPOINT=XXXXXX
export GENERIC_AUTHORIZATION_ENDPOINT=XXXXXX
```

I'm using Keycloak OpenID connect client. (it's a fresh Keycloak install)

The traceback is : 
```bash
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/usr/local/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/usr/local/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 274, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.9/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/usr/local/lib/python3.9/site-packages/litellm/proxy/proxy_server.py", line 5367, in auth_callback
    user_id = getattr(result, "first_name", "") + getattr(result, "last_name", "")
TypeError: unsupported operand type(s) for +: 'NoneType' and 'NoneType'
```

There was also before that a small warning message : 
```bash
No response convertor was provided, returned OpenID will always be empty
```

Looking into [the FastAPI SSO plugin](https://github.com/tomasvotava/fastapi-sso/blob/master/fastapi_sso/sso/generic.py#L67-L73) I have seen that a `response_converter` is required to parse the response from the SSO UserInfo endpoint. This endpoint is used to retreiv information about the user : first_name, last_name, email, ... (the information depends on Scopes defined during the initial query)

I have tried to modify the Litellm proxy code to add a converter :
```python
from fastapi_sso.sso.base import OpenID
def convert_openid(response, _client):
 """Convert user information returned by OIDC"""
    print(response)
    return OpenID(display_name=response["sub"], id=response["preferred_username"], email=response["email"], first_name=response["first_name"], last_name=response["last_name"])

SSOProvider = create_provider(name="oidc", discovery_document=discovery, response_convertor=convert_openid)
```

**The values that I have put inside the `convert_openid` are specific to my SSO config. Values from `GENERIC_*_ATTRIBUTE` should be used.**

With this code modification, the code run better and I need to try make the full connexion process on my deployment to make sure this is the only issue with the code

**Comment by Lunik** [comment]
#2129 seems to resolve part of the issue

**Comment by ishaan-jaff** [comment]
Call notes:
- Error was caused because we don't have a response convertor 
- Merging #2129 would solve this issue (this PR solves it well since they maintain generic_user_name, generic_user_email attributes)

Action item:
- merge #2129

**Comment by ishaan-jaff** [comment]
I set up keycloak locally this is the response I got fro key cloak:
```
 id='ishaan@berri.ai' email='ishaan@berri.ai' first_name=None last_name=None display_name='ishaan@berri.ai' picture=None provider=None
 ```

**Comment by ishaan-jaff** [comment]
Merged #2129 needed to make minor changes for fully enable key cloak https://github.com/BerriAI/litellm/pull/2140
